google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/impl/GoogleCloudStorageClient.scala

```diff
@@ -0,0 +1,459 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage.impl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.Uri.{Path, Query}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{`Content-Range`, OAuth2BearerToken, RawHeader}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import akka.stream.alpakka.googlecloud.storage.GoogleCloudStorageModel._
+import akka.stream.alpakka.googlecloud.storage.StorageSettings
+import akka.stream.alpakka.googlecloud.storage.impl.Formats._
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import akka.util.ByteString
+import akka.{Done, NotUsed}
+import spray.json._
+
+import scala.collection.immutable.Seq
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+
+@InternalApi
+private[storage] object GoogleCloudStorageClient {
+  def apply(settings: StorageSettings)(implicit system: ActorSystem, mat: Materializer): GoogleCloudStorageClient = {
+    val tokenApiSettings: TokenApiSettings = TokenApiSettings(
+      url = "https://www.googleapis.com/oauth2/v4/token",
+      scope = "https://www.googleapis.com/auth/devstorage.read_write"
+    )
+
+    new GoogleCloudStorageClient(settings) {
+      val baseUri: Uri = Uri("https://www.googleapis.com/")
```

Comment:
There are no official tools, I found this one https://github.com/teone/gc-fake-storage, but I'm not sure does it make sense to use it since we use wiremock to mock request and responses and we are in controle of it. If something changes in google api we would need to wait for update from `gc-fake-storage` to test it properly...